p2p/node/p2p_services.go

```diff
@@ -77,9 +78,22 @@ func (p *P2PNode) requestFromPeer(peerID peer.ID, location common.Location, data
 	// Check the received data type & hash matches the request
 	switch datatype.(type) {
 	case *types.Block:
-		if block, ok := recvdType.(*types.Block); ok && block.Hash() == data.(common.Hash) {
-			return block, nil
+		if block, ok := recvdType.(*types.Block); ok {
+			switch data := data.(type) {
+			case common.Hash:
+				if block.Hash() == data {
+					return block, nil
+				}
+				return nil, errors.Errorf("invalid response: expected block with hash %s, got %s", data, block.Hash())
+			case *big.Int:
```

Comment:
I don't understand. Why would we receive a block number when we asked for a block?

Reply:
ahhh, I thought you were switching on recvdType... got it. In that case this code looks good. Sorry for the confusion :heavy_check_mark: